packages/url/src/test/index.test.js

```diff
@@ -303,17 +325,18 @@ describe( 'isValidFragment', () => {
 		expect( isValidFragment( '#yes_it_is' ) ).toBe( true );
 		expect( isValidFragment( '#yes~it~is' ) ).toBe( true );
 		expect( isValidFragment( '#yes-it-is' ) ).toBe( true );
+		expect( isValidFragment( '#yes-it-is?' ) ).toBe( true );
+		expect( isValidFragment( '#yes-it-is?not-a-query' ) ).toBe( true );
+		expect( isValidFragment( '#yes-it-is#' ) ).toBe( true );
+		expect( isValidFragment( '#yes-it-#is' ) ).toBe( true );
```

Comment:
I'm surprised these are valid fragments (containing hashes)

Reply:
Me too, but that's what the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) API returns in Firefox.